shared/ui-components/src/modals/Modal.tsx

```diff
@@ -26,46 +32,97 @@ export const Modal: FunctionComponent<ModalProps> = ({
   fullScreen = false,
   onClose,
   children,
+  ariaDescription,
 }) => {
+  // append modal to the root of app
+  const modalRoot = typeof document !== "undefined" ? document.querySelector("#__next") : null
+  const el = typeof document !== "undefined" ? document.createElement("div") : null
+  const siteContainer =
+    typeof document !== "undefined" ? document.querySelector(".site-container") : null
+
+  const isEscClicked = useKeyPress("Escape")
+
+  const modalContentRef = createRef<HTMLDivElement>()
+
+  useEffect(() => {
+    if (!modalRoot || !el || !siteContainer || !open) return
+    modalRoot.appendChild(el)
+    siteContainer.setAttribute("aria-hidden", "true")
+    disableBodyScroll(el)
+
+    return () => {
+      siteContainer.removeAttribute("aria-hidden")
+      enableBodyScroll(el)
+      modalRoot.removeChild(el)
+    }
+  }, [el, modalRoot, siteContainer, open])
+
+  useEffect(() => {
+    if (isEscClicked && onClose) {
+      onClose()
+    }
+  }, [isEscClicked, onClose])
```

Comment:
I think it makes a little bit more sense to me to have `useKeyPress` take a `onKeyPress` handler instead of requiring us to define another effect here....

Reply:
I refactored this hook to do not store state, I think it will be simpler & enough for most purposes.